src/meta/processors/indexMan/CreateEdgeIndexProcessor.cpp

```diff
@@ -99,18 +102,35 @@ void CreateEdgeIndexProcessor::process(const cpp2::CreateEdgeIndexReq& req) {
 
     const auto& schemaCols = latestEdgeSchema.get_columns();
     std::vector<cpp2::ColumnDef> columns;
-    for (auto &field : fieldNames) {
+    for (auto &field : fields) {
         auto iter = std::find_if(schemaCols.begin(), schemaCols.end(),
-                                 [field](const auto& col) { return field == col.get_name(); });
+                                 [field](const auto& col) {
+                                     return field.get_name() == col.get_name();
+                                 });
 
         if (iter == schemaCols.end()) {
-            LOG(ERROR) << "Field " << field << " not found in Edge " << edgeName;
+            LOG(ERROR) << "Field " << field.get_name() << " not found in Edge " << edgeName;
             handleErrorCode(cpp2::ErrorCode::E_NOT_FOUND);
             onFinished();
             return;
-        } else {
-            columns.emplace_back(*iter);
         }
+        cpp2::ColumnDef col = *iter;
+        if (col.type.get_type() == meta::cpp2::PropertyType::STRING) {
+            if (!field.__isset.type_length) {
+                LOG(ERROR) << "No type length set : " << field.get_name();
+                handleErrorCode(cpp2::ErrorCode::E_INVALID_PARM);
+                onFinished();
+                return;
+            }
+            col.type.set_type(meta::cpp2::PropertyType::FIXED_STRING);
+            col.type.set_type_length(*field.get_type_length());
+        } else if (field.__isset.type_length) {
+            LOG(ERROR) << "No need to set type length : " << field.get_name();
+            handleErrorCode(cpp2::ErrorCode::E_INVALID_PARM);
+            onFinished();
+            return;
+        }
```

Comment:
If it is a non-string type, must an error be reported when type_length is set?

Reply:
> If it is a non-string type, must an error be reported when type_length is set?

yeah，if is an no-string type, don't need to set the length. else an error code will be sent.